src/LogicApp/LogicApp/Cmdlets/IntegrationAccount/NewAzureIntegrationAccountAssemblyCommand.cs

```diff
@@ -0,0 +1,176 @@
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.LogicApp.Cmdlets
+{
+    using Microsoft.Azure.Commands.LogicApp.Utilities;
+    using Microsoft.Azure.Commands.ResourceManager.Common;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+    using Microsoft.Azure.Management.Logic.Models;
+    using Microsoft.WindowsAzure.Commands.Utilities.Common;
+    using System.Collections;
+    using System.Management.Automation;
+
+    /// <summary>
+    /// Creates a new integration account assembly.
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, AzureRMConstants.AzureRMPrefix + "IntegrationAccountAssembly", DefaultParameterSetName = ParameterSet.ByIntegrationAccountAndFilePath, SupportsShouldProcess = true)]
+    [OutputType(typeof(AssemblyDefinition))]
+    public class NewAzureIntegrationAccountAssemblyCommand : LogicAppBaseCmdlet
+    {
+        #region Input Parameters
+
+        [Parameter(Mandatory = true, HelpMessage = Constants.ResourceGroupHelpMessage, ParameterSetName = ParameterSet.ByIntegrationAccountAndContentLink)]
+        [Parameter(Mandatory = true, HelpMessage = Constants.ResourceGroupHelpMessage, ParameterSetName = ParameterSet.ByIntegrationAccountAndFileBytes)]
+        [Parameter(Mandatory = true, HelpMessage = Constants.ResourceGroupHelpMessage, ParameterSetName = ParameterSet.ByIntegrationAccountAndFilePath)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = Constants.IntegrationAccountNameHelpMessage, ParameterSetName = ParameterSet.ByIntegrationAccountAndContentLink)]
+        [Parameter(Mandatory = true, HelpMessage = Constants.IntegrationAccountNameHelpMessage, ParameterSetName = ParameterSet.ByIntegrationAccountAndFileBytes)]
+        [Parameter(Mandatory = true, HelpMessage = Constants.IntegrationAccountNameHelpMessage, ParameterSetName = ParameterSet.ByIntegrationAccountAndFilePath)]
+        [ResourceNameCompleter("Microsoft.Logic/integrationAccounts", nameof(ResourceGroupName))]
+        [ValidateNotNullOrEmpty]
+        [Alias("IntegrationAccountName")]
+        public string ParentName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = Constants.AssemblyNameHelpMessage, ParameterSetName = ParameterSet.ByInputObjectAndContentLink)]
```

Comment:
If this is mandatory and the help message is the same across all parameter sets you can just have:
```
        [Parameter(Mandatory = true, HelpMessage = Constants.AssemblyNameHelpMessage)]
        [ResourceNameCompleter("Microsoft.Logic/integrationAccounts/assemblies", nameof(ResourceGroupName), nameof(ParentName))]
        [ValidateNotNullOrEmpty]
        [Alias("AssemblyName", "ResourceName")]
        public string Name { get; set; }
```
This will look a little neater.  Apply to other places where the parameter is in all parameter sets.

Reply:
I think it's just the name in the 2 creation cmdlets. Changed